packages/access-client/src/agent.js

```diff
@@ -191,19 +253,36 @@ export class Agent {
   }
 
   /**
-   * Get all the proofs matching the capabilities
+   * Get all the proofs matching the capabilities.
+   *
+   * Proofs are delegations with an audience matching agent DID, or with an
+   * audience matching the session DID.
    *
-   * Proofs are delegations with an audience matching agent DID.
+   * Proof of session will also be included in the returned proofs if any
+   * proofs matching the passed capabilities require it.
    *
    * @param {import('@ucanto/interface').Capability[]} [caps] - Capabilities to filter by. Empty or undefined caps with return all the proofs.
    */
   proofs(caps) {
     const arr = []
+    const session = this.#data.sessionProof()
```

Comment:
seems like we could accomodate multiple sessionProofs here. We'd just need to index them by `.audience.did()`.
Might also be good to pass `caps` to sessionProof to it can omit any that aren't relevant.

Reply:
I removed `sessionProof` from the public api so idc what we do here for now